src/wp-includes/class-wp-plugin-dependencies.php

```diff
@@ -0,0 +1,884 @@
+			self::$dependencies[ $plugin ] = $dependency_slugs;
+			self::$dependency_slugs        = array_merge( self::$dependency_slugs, $dependency_slugs );
+		}
+
+		$dependent_keys         = array_keys( self::$dependencies );
+		self::$dependent_slugs  = array_combine( $dependent_keys, array_map( 'dirname', $dependent_keys ) );
+		self::$dependency_slugs = array_unique( self::$dependency_slugs );
+	}
+
+	/**
+	 * Sanitizes slugs.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string $slugs A comma-separated string of plugin dependency slugs.
+	 * @return array An array of sanitized plugin dependency slugs.
+	 */
+	protected static function sanitize_dependency_slugs( $slugs ) {
+		$sanitized_slugs = array();
+		$slugs           = explode( ',', $slugs );
+
+		foreach ( $slugs as $slug ) {
+			$slug = trim( $slug );
+
+			/**
+			 * Filters a plugin dependency's slug before matching to
+			 * the WordPress.org slug format.
+			 *
+			 * Can be used to switch between free and premium plugin slugs, for example.
+			 *
+			 * @since 6.4.0
+			 *
+			 * @param string $slug The slug.
+			 */
+			$slug = apply_filters( 'wp_plugin_dependencies_slug', $slug );
+
+			// Match to WordPress.org slug format.
+			if ( preg_match( '/^[a-z0-9]+(-[a-z0-9]+)*$/mu', $slug ) ) {
+				$sanitized_slugs[] = $slug;
+			}
+		}
+		$sanitized_slugs = array_unique( $sanitized_slugs );
+		sort( $sanitized_slugs );
+
+		return $sanitized_slugs;
+	}
+
+	/**
+	 * Gets the slugs of plugins that the dependent requires.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string $plugin_file The dependent plugin's filepath, relative to the plugins directory.
+	 * @return array An array of dependency plugin slugs.
+	 */
+	protected static function get_dependencies( $plugin_file ) {
+		if ( isset( self::$dependencies[ $plugin_file ] ) ) {
+			return self::$dependencies[ $plugin_file ];
+		}
+
+		return array();
+	}
+
+	/**
+	 * Gets filepaths of plugins that require the dependency.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string $slug The dependency's slug.
+	 * @return array An array of dependent plugin filepaths, relative to the plugins directory.
+	 */
+	protected static function get_dependents( $slug ) {
+		$dependents = array();
+
+		foreach ( self::$dependencies as $dependent => $dependencies ) {
+			if ( in_array( $slug, $dependencies, true ) ) {
+				$dependents[] = $dependent;
+			}
+		}
+
+		return $dependents;
+	}
+
+	/**
+	 * Gets plugin filepaths for active plugins that depend on the dependency.
+	 *
+	 * Recurses for each dependent that is also a dependency.
+	 *
+	 * @param string $plugin_file The dependency's filepath, relative to the plugin directory.
+	 * @return string[] An array of active dependent plugin filepaths, relative to the plugin directory.
+	 */
+	protected static function get_active_dependents_in_dependency_tree( $plugin_file ) {
+		$all_dependents = array();
+		$slug           = str_contains( $plugin_file, '/' ) ? basename( dirname( $plugin_file ) ) : $plugin_file;
```

Comment:
No need to use `basename()` here. Everything is stated to be relative to the plugins directory.
```suggestion
		$slug           = str_contains( $plugin_file, '/' ) ? dirname( $plugin_file ) : $plugin_file;
```